### ISSUE TITLE
Automatic resource tracking

### ISSUE THREAD

**Issue by wainwrightmark** [johanhelsing/bevy_pkv#42]
Hi. 
Thanks for making this plugin - it's unbelievably useful.

For my game I've built quite an ergonomic wrapper around it and I think might be worth adding something like it to this crate.
Basically, you call `app.init_tracked_resource<T>()` for some `Resource`, and then that resource gets initialized with the value in the `PkvStore` and every time the resource changes, the store updates. You never need to use the `PkvStore` directly.

Happy to contribute code / examples / documentation if this is considered worth adding.

**Comment by JeanMertz** [comment]
@wainwrightmark any chance of sharing (part of) your implementation here for others to use, even if it isn't upstreamed into `bevy_pkv`?

**Comment by johanhelsing** [comment]
Yes sounds very useful. Happy to accept a pr if the implementation is clean. Sorry i forgot to respond to this
